File: app/src/main/java/se/kicksort/metalarchives/band/AlbumViewHolder.java
package se.kicksort.metalarchives.band;

import com.github.wrdlbrnft.sortedlistadapter.SortedListAdapter;

import io.reactivex.Observable;
import io.reactivex.subjects.PublishSubject;
import se.kicksort.metalarchives.databinding.AlbumListEntryBinding;
import se.kicksort.metalarchives.model.TinyAlbum;

public class AlbumViewHolder extends SortedListAdapter.ViewHolder<TinyAlbum> {
    private final AlbumListEntryBinding binding;
    private PublishSubject<TinyAlbum> clickSubject = PublishSubject.create();

    public AlbumViewHolder(AlbumListEntryBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
        this.binding.getRoot().setOnClickListener(view -> clickSubject.onNext(this.binding.getAlbum()));
    }

    @Override
    public void performBind(TinyAlbum item) {
        binding.setAlbum(item);
    }

    public Observable<TinyAlbum> getClicks() {
        return clickSubject;
    }
}

File: app/src/main/java/se/kicksort/metalarchives/band/AlbumViewHolder.kt
package se.kicksort.metalarchives.band

import com.github.wrdlbrnft.sortedlistadapter.SortedListAdapter

import io.reactivex.Observable
import io.reactivex.subjects.PublishSubject
import se.kicksort.metalarchives.databinding.AlbumListEntryBinding
import se.kicksort.metalarchives.model.TinyAlbum

class AlbumViewHolder(private val binding: AlbumListEntryBinding) : SortedListAdapter.ViewHolder<TinyAlbum>(binding.root) {
    private val clickSubject = PublishSubject.create<TinyAlbum>()

    val clicks: Observable<TinyAlbum>
        get() = clickSubject

    init {
        this.binding.root.setOnClickListener { view -> clickSubject.onNext(this.binding.album) }
    }

    public override fun performBind(item: TinyAlbum) {
        binding.album = item
    }
}
